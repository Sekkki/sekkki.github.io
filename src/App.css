.App {
  background-color: $main;
  text-align: center;
}

#root {
  min-height: 100%;
  display: flex;
  position: relative;
}

.root {
  min-height: 100%;
  height: 2000px;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.root {
  flex-grow: 1;
  min-height: 100%;
  display: flex;
  justify-content: center;
  flex-direction: column;
}

.container-height {
  flex: 1 0 auto;
  flex-grow: 1;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
