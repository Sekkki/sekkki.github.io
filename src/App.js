import React from 'react';
import { BrowserRouter as Router, Switch, Route, Redirect, BrowserRouter } from "react-router-dom";
import './App.css';
import 'semantic-ui-css/semantic.min.css'
import { withCookies } from 'react-cookie';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import reducers from './reducers';
import Layout from './components/Layouts/index';
// import adminLayout from './components/Layouts/admin/layout';
import Main from './components/blocks/main/index';

const middleware = applyMiddleware(thunk)(createStore);

const store = middleware(reducers);

class App extends React.Component {

  render() {

    const AppRoute = ({ component: Component, layout: Layout, ...rest }) => (
      <Route {...rest} render={props => (
        <Layout>
          <Component {...props} />
        </Layout>
      )} />
    )
    
    /* const PrivateRoute = ({ component: Component, layout: Layout, ...rest }) => (
      <Route {...rest} render={(props) => (
        store.isAuthenticated === true
          ? <Layout store={store}><Component {...props} /></Layout>
          : <Redirect to='/login' />
      )} />
    ) */
    return (
      <Provider store={store}>
        <BrowserRouter>
          <Router>
            <AppRoute exact path="/" layout={Layout} component={() => <Main />} />         
          </Router> 
        </BrowserRouter>
      </Provider>
    );
  }
}

export default withCookies(App);
