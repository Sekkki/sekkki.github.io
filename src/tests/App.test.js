import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import renderer from 'react-test-renderer';

describe('<App />', () => {
it('renders without crashing', () => {
  const component = renderer
    .create(
      <App />
    )
    .toJSON();
  expect(component).toMatchSnapshot();
});
})