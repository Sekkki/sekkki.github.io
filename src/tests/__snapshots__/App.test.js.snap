// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders without crashing 1`] = `
<div
  className="root"
>
  <header
    className="header"
  >
    <div
      className="brand"
    >
      <p>
        Hyppnotic
        <span>
          store.
        </span>
      </p>
    </div>
    <div
      className="menu"
    >
      <ul>
        <li>
          Каталог
        </li>
        <li>
          О нас
        </li>
        <li>
          Помощь
        </li>
        <li>
          Поддержка
        </li>
      </ul>
    </div>
    <div
      className="helper-wrapper"
    >
      <div
        className="helper-container"
      >
        <div
          className="helper"
          onClick={[Function]}
        >
          <i
            className="fas fa-arrow-left"
          />
        </div>
        <div
          className="helper-button"
        >
          <i
            className="fas fa-sign-in-alt"
          />
          <i
            className="fas fa-palette"
          />
          <i
            className="fas fa-ruble-sign"
          />
          <i
            className="fas fa-balance-scale"
          />
        </div>
      </div>
    </div>
    <div
      className="mobile-menu-container"
    >
      <div
        className="mobile-menu"
      >
        <div
          className="mobile-menu-brand"
        >
          Hyppnotic
        </div>
        <div
          className="mobile-menu-button"
          onClick={[Function]}
        >
          <i
            className="fas fa-times"
          />
        </div>
        <div
          className="mobile-menu-links"
        >
          Links
        </div>
      </div>
    </div>
    <div
      className="button-xs"
      onClick={[Function]}
    >
      <i
        className="fas fa-bars"
      />
    </div>
  </header>
  <div
    className="container-height"
  >
    <div>
      Main
    </div>
  </div>
  <footer>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12 col-12"
        >
          <div
            className="footer-brand"
          >
            Hyppnotic.store
          </div>
        </div>
        <div
          className="col-lg-6 col-md-6 col-12"
        >
          <div
            className="footer-links"
          >
            <ul>
              <li>
                Главная
              </li>
              <li>
                Магазин
              </li>
              <li>
                О нас
              </li>
              <li>
                Помощь
              </li>
              <li>
                Поддержка
              </li>
            </ul>
          </div>
        </div>
        <div
          className="col-lg-6 col-md-6 col-12"
        >
          <div
            className="footer-links"
          >
            <ul>
              <li>
                Группа VK
              </li>
              <li>
                Instagram
              </li>
              <li>
                Twitter
              </li>
              <li>
                Github
              </li>
              <li>
                Facebook
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      className="footer-rights container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <p>
            All rights are reserved by Hyppnotic company @2018-2019
          </p>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
